hazenlib/snr.py

```diff
@@ -171,32 +232,40 @@ def get_object_centre(dcm) -> (int, int):
 
     """
 
-    shape_detector = hazenlib.tools.ShapeDetector(arr=dcm.pixel_array)
-    orientation = hazenlib.tools.get_image_orientation(dcm.ImageOrientationPatient)
-
-    if orientation in ['Sagittal', 'Coronal']:
-        # orientation is sagittal to patient
-        try:
-            (col, row), size, angle = shape_detector.get_shape('rectangle')
-        except exc.ShapeError:
-            # shape_detector.find_contours()
-            # shape_detector.detect()
-            # contour = shape_detector.shapes['rectangle'][1]
-            # angle, centre, size = cv.minAreaRect(contour)
-            # print((angle, centre, size))
-            # im = cv.drawContours(dcm.pixel_array.copy(), [shape_detector.contours[0]], -1, (0, 255, 255), 10)
-            # plt.imshow(im)
-            # plt.savefig("rectangles.png")
-            # print(shape_detector.shapes.keys())
-            raise
-    elif orientation == 'Transverse':
-        try:
-            col, row, r = shape_detector.get_shape('circle')
-        except exc.MultipleShapesError:
-            print('Warning! Found multiple circles in image, will assume largest circle is phantom.')
-            col, row, r = get_largest_circle(shape_detector.shapes['circle'])
-    else:
-        raise Exception("Direction must be Transverse, Sagittal or Coronal.")
+    # Shape Detection
+    try:
+        shape_detector = hazenlib.tools.ShapeDetector(arr=dcm.pixel_array)
+        orientation = hazenlib.tools.get_image_orientation(dcm.ImageOrientationPatient)
+
+        if orientation in ['Sagittal', 'Coronal']:
+            # orientation is sagittal to patient
+            try:
+                (col, row), size, angle = shape_detector.get_shape('rectangle')
+            except exc.ShapeError:
+                # shape_detector.find_contours()
+                # shape_detector.detect()
+                # contour = shape_detector.shapes['rectangle'][1]
+                # angle, centre, size = cv.minAreaRect(contour)
+                # print((angle, centre, size))
+                # im = cv.drawContours(dcm.pixel_array.copy(), [shape_detector.contours[0]], -1, (0, 255, 255), 10)
+                # plt.imshow(im)
+                # plt.savefig("rectangles.png")
+                # print(shape_detector.shapes.keys())
+                raise
+        elif orientation == 'Transverse':
+            try:
+                col, row, r = shape_detector.get_shape('circle')
+            except exc.MultipleShapesError:
+                print('Warning! Found multiple circles in image, will assume largest circle is phantom.')
+                col, row, r = get_largest_circle(shape_detector.shapes['circle'])
+        else:
+            raise exc.ShapeError("Unable to identify phantom shape.")
+
+    # Threshold Detection
+    except exc.ShapeError:
+        # print('Shape Detection failed. Performing Thresholding ...')
```

Comment:
Can you add a logger statement that explains whats happening at the various points during this function? I think most login can be debug level but maybe add an info logging statement for when the shape detector fails and the thresholding method is used?

You can merge in (or just have a look at) origin/feature/HZN-80---adding-logging-to-screen to see how the logger will wor but basically you just put something like

`logger.info('this is an info logging statement that shows up all the time because info is the default level')`

`logging.debug('this is a debug statement that will show up when someone asked for more detailed logs')`